simulator/func_sim/alu.h

```diff
@@ -213,8 +213,20 @@ struct ALU
     template<typename I> static void xnor( I* instr)  { instr->v_dst[0] = instr->v_src[0] ^ ~instr->v_src[1]; }
 
     // Bit permutation
+    
     template<typename I> static void grev( I* instr) { instr->v_dst[0] = gen_reverse( instr->v_src[0], shamt_v_src2<typename I::RegisterUInt>( instr)); }
-
+    
+    template<typename I> static
+    void riscv_shfl( I* instr)
+    {
+        auto dst_value = instr->v_src[0];
+        constexpr size_t limit = log_bitwidth<decltype( instr->v_src[0])> - 1;
+            for ( size_t i = limit ; i > 0; --i)
```

Comment:
This nesting is redundant

Reply:
How can I fix it?

Reply:
I need to commit opcodes in mipt-mips repository?

Reply:
I put a sequence of commands in a comment [above](https://github.com/MIPT-ILab/mipt-mips/pull/1398#pullrequestreview-526151382). Please check.

Reply:
I asced abou it:" This nesting is redundant "

Reply:
1) How to return ion branch without my commits
2)to update  opcodes I need to be in mipt-mips/riscv-opcodes or in separate directory?

Reply:
1. What is "ion branch"? Why do you need a branch without your commits?
2. I gave you detailed instructions on updating submodule version.

Reply:
https://github.com/MIPT-ILab/mipt-mips/pull/1398#pullrequestreview-526151382

Reply:
1)To do new corrected commit

Reply:
Just do new commits fixing nesting and submodule version. GitHub will automatically append it to this pull request.

Reply:
> GitHub will automatically append it to this pull request.

*once you push them to your fork